sdcflows/interfaces/bspline.py

```diff
@@ -123,16 +127,45 @@ class BSplineApprox(SimpleInterface):
 
     def _run_interface(self, runtime):
         from sklearn import linear_model as lm
-        from scipy.sparse import vstack as sparse_vstack
+
+        # Output name baseline
+        out_name = fname_presuffix(
+            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        )
 
         # Load in the fieldmap
         fmapnii = nb.load(self.inputs.in_data)
+        zooms = fmapnii.header.get_zooms()
+
+        # Get a mask (or define on the spot to cover the full extent)
+        masknii = (
+            nb.load(self.inputs.in_mask)
+            if isdefined(self.inputs.in_mask)
+            else None
+        )
+
+        need_resize = np.any(np.array(zooms) < self.inputs.zooms_min)
+        if need_resize:
+            from niworkflows.utils.images import resample_by_spacing
+
+            LOGGER.info(
+                "Resampling image with resolution exceeding 'zooms_min' "
+                f"({'x'.join(str(s) for s in zooms)})."
+            )
+            fmapnii = resample_by_spacing(fmapnii, [self.inputs.zooms_min] * 3)
+
+            if masknii is not None:
+                masknii = resample_by_spacing(masknii, [self.inputs.zooms_min] * 3)
```

Comment:
Here we'd need to do something slightly different. Not positive if blur-downsample-(threshold at 0.5) is equivalent to a nearest-neighbor. You're also not doing a nearest-neighbor here, so it's a bug whatever downsampling method we take.

Reply:
Not sure I follow. This is going to interpolate at the edges of the binary mask (if binary, it doesn't have to be), so you will have some "spreadout" from the bspline kernel, but I don't think this is going to be problematic tbh.

Then, another question is whether the threshold for binarization in line 165 is too low.

Reply:
I thought the mask was binary. I'll need to look again.